### PR TITLE
Create a new query request for each Query

### PR DESCRIPTION
**Pull request by sparrc**
fixes https://github.com/influxdb/telegraf/issues/351

### CODE REVIEW

**Comment by sparrc** [comment]
We were modifying the URL object, which gets used for queries and writes. Thus when we were writing we were still getting the last query used in the write call.

File: client/v2/client.go [hunk 1]
```diff
@@ -426,25 +425,26 @@ func (uc *udpclient) Query(q Query) (*Response, error) {
...
 	if err != nil {
 		return nil, err
 	}
+	req.Header.Set("Content-Type", "")
 	req.Header.Set("User-Agent", c.useragent)
 	if c.username != "" {
 		req.SetBasicAuth(c.username, c.password)
 	}
 
+	params := req.URL.Query()
```
Comment (otoolep) [inline client/v2/client.go]: I don't really understand how this change fixes anything.

Comment (sparrc) [inline client/v2/client.go]: we get a new `params` object from the specific request that we are making.

before, we were directly modifying `values` of the structs `URL` object

Comment (otoolep) [inline client/v2/client.go]: OK, I am tired, I see the difference now.

**Comment by otoolep** [comment]
+1

**Comment by JulienChampseix** [comment]
is it fixed or merged on the master ? cuz not yet present into ChangeLog file on the master

**Comment by otoolep** [comment]
As you can see this PR has been marked as merged, so yes, it is present in master.

**Pull request merged**